project.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+
+# Load constant
+source .env
+
+# Define output formats
+question=$(tput bold)
+error=$(tput setaf 1)
+info=$(tput setaf 3)
+success=$(tput setaf 4)
+reset=$(tput sgr0)
+qe=$(tput bold && tput setaf 1)
+re=$(tput sgr0 && tput setaf 1)
+
+############
+# Function #
+############
+function new() {
+    echo "${info}New started...${reset}"
+
+    # Initialize project
+    rm -rf $PROJECT_DIR
+    git clone https://gitlab.omn.proximis.com/public-packages/change-new-project $PROJECT_DIR
+
+    # Update composer
+    composer self-update
+    composer update --working-dir=$PROJECT_DIR --prefer-dist --ignore-platform-reqs
+
+    echo "${info}Do not forget to launch init.sh script and to create the file project.local.json in App/Config before executing update.sh script${reset}"
+}
+
+function bootstrap() {
+    echo "${info}Bootstrap started...${reset}"
+
+    # Clone project
+    rm -rf $PROJECT_DIR
+    git clone ssh://git@gitlab.omn.proximis.com:10022/$INTEGRATOR/$PROJECT.git $PROJECT_DIR
+
+    # Add forked remote
+    git --git-dir=$GIT_DIR --work-tree=$GIT_WORK_TREE checkout -b dev origin/dev
+    git --git-dir=$GIT_DIR --work-tree=$GIT_WORK_TREE remote add composer ssh://git@gitlab.omn.proximis.com:10022/$REMOTE/$PROJECT.git
+    git --git-dir=$GIT_DIR --work-tree=$GIT_WORK_TREE fetch composer
+
+    # TODO : Add project.local.json in a better way
+    git clone git@bitbucket.org:snippets/emakinafr-commerce/re8zR5/projectlocaljson.git $PROJECT_DIR/App/Config/project-local-json
+    mv $PROJECT_DIR/App/Config/project-local-json/project.local.json $PROJECT_DIRECTORY/App/Config/project.local.json
+    rm -rf $PROJECT_DIRECTORY/App/Config/project-local-json
+
+    # Adding the nginx configuration to the docker project
+    NGINX_DIR="$PROJECT_DIR/nginx"
+    NGINX_CONFIGS=$(ls $NGINX_DIR 2> /dev/null)
+    if [[ ! -z $NGINX_CONFIGS ]]; then
+        for NGINX_CONFIG in $NGINX_CONFIGS; do
+            cp $NGINX_DIR/$NGINX_CONFIG ./nginx/site-available/$NGINX_CONFIG
+        done 
+    fi
+
+    # Check if the project is a directory
+    if [[ ! -d $PROJECT_DIR ]]; then
+        echo "${error}The ${qe}project directory${re} does not exist.${reset}"
+        exit 1
+    else
+        echo "${success}Bootstrap completed successfully.${reset}"
+    fi
+
+    update
+}
+
+function update() {
+    echo "${info}Update started...${reset}"
+
+    # Remove all plugins
+    rm -rf $PROJECT_DIR/Plugins/*
+
+    # Update composer
+    composer self-update
+    composer update --working-dir=$PROJECT_DIR --prefer-dist --ignore-platform-reqs
```

Comment:
Is it safe enough to perform an update with the `--ignore-platform-reqs` parameter? Maybe you can run the command inside the container instead.

Reply:
We do it to be able to feed the plugin folder and after install plugin from GitLab